src/js/utils/html-utils.js

```diff
@@ -0,0 +1,145 @@
+         * is returned with the original text escaped. If the input
+         * is already an HtmlSnippet then it is known to be already
+         * escaped and so it is just returned directly.
+         *
+         * @param {(string|HtmlSnippet)} text Some HTML that is
+         * either provided as a string or as an HtmlSnippet.
+         * @returns {HtmlSnippet} A safely escaped HtmlSnippet.
+         */
+        escape = function(text) {
+            if (text instanceof HtmlSnippet) {
+                return text;
+            } else {
+                return HTML(_.escape(text));
+            }
+        };
+
+        /**
+         * Returns an HTML snippet by interpolating the provided parameters.
+         *
+         * The HTML text is provided as a tokenized format string where parameters
+         * are indicated via curly braces, e.g. 'Hello {name}'. These tokens are
+         * replaced by the parameter value of the same name.
+         *
+         * Parameter values will be rendered using their toString methods and then
+         * HTML-escaped. The only exception is that instances of the class HTML
+         * are rendered without escaping as their contract declares that they are
+         * already valid HTML.
+         *
+         * Example:
+         *   HtmlUtils.interpolateHtml(
+         *       'You are enrolling in {spanStart}{courseName}{spanEnd}',
+         *       {
+         *           courseName: 'Rock & Roll 101',
+         *           spanStart: HtmlUtils.HTML('<span class="course-title">'),
+         *           spanEnd: HtmlUtils.HTML('</span>')
+         *       }
+         *   );
+         *
+         * returns:
+         *   'You are enrolling in <span class="course-title">Rock &amp; Roll 101</span>'
+         *
+         * Note: typically the formatString will need to be internationalized, in which
+         * case it will be wrapped with a call to an i18n lookup function. In Django,
+         * this would look like:
+         *
+         *   HtmlUtils.interpolateHtml(
+         *       gettext('You are enrolling in {spanStart}{courseName}{spanEnd}'),
+         *       ...
+         *   );
+         *
+         * @param {string} formatString The string to be interpolated.
+         * @param {Object} parameters An optional set of parameters to the template.
+         * @returns {HtmlSnippet} The resulting safely escaped HTML snippet.
+         */
+        interpolateHtml = function(formatString, parameters) {
+            var result = StringUtils.interpolate(
+                escape(formatString).toString(),
+                _.mapObject(parameters, escape)
+            );
+            return HTML(result);
+        };
+
+        /**
+         * Returns a function that renders an Underscore template as an HTML snippet.
+         *
+         * Note: This helper function makes the following context parameters
+         * available to the template in addition to those passed in:
```

Comment:
Are there supposed to be multiple parameters? I'm having trouble understanding the documentation.